Adjust text formatting per CJK copywriting guidelines

ihavecoke opened the pull request:
### Description

Refined text formatting to follow the recommendations from: https://sparanoid.com/note/chinese-copywriting-guidelines/

### Motivation

This change aims to improve the readability and professionalism of the copy by following the best practices outlined in the CJK Copywriting Guidelines. Specifically, it addresses:

- Adding spaces between English and Chinese words
- Adding spaces between numbers and units
- Using fullwidth punctuation for Chinese text
- Using halfwidth Arabic numerals
- Using halfwidth punctuation for English sentences
- Using correct casing for proper nouns
- Avoiding jargons and abbreviations

These adjustments will enhance the overall reading experience for both Chinese and English audiences.

### Additional details

The changes are based on the guidelines provided at https://sparanoid.com/note/chinese-copywriting-guidelines/, which is a widely recognized resource for Chinese copywriting best practices.

Josh-Cena reviewed (changes requested):
cc @mdn/yari-content-zh @mdn/yari-content-ja @mdn/yari-content-ko 

At the very least, @ihavecoke please revert any changes to example code output, unless the browsers actually output the current text.

yin1999 commented on files/en-us/web/javascript/guide/text_formatting/index.md:
```diff
@@ -292,7 +292,7 @@ const hanDecimalRMBInChina = new Intl.NumberFormat("zh-CN-u-nu-hanidec", {
 });
 
-console.log(hanDecimalRMBInChina.format(1314.25)); // ￥ 一,三一四.二五
+console.log(hanDecimalRMBInChina.format(1314.25)); // ￥ 一，三一四。二五
```
The original comment is correct:

![image](https://github.com/mdn/content/assets/15844309/808687d3-e32e-4bfd-9768-e0974f59b4f1)

yin1999 commented on files/en-us/web/javascript/reference/global_objects/bigint/tolocalestring/index.md:
```diff
@@ -100,7 +100,7 @@ console.log(bigint.toLocaleString("en-IN"));
 // the nu extension key requests a numbering system, e.g. Chinese decimal
 console.log(bigint.toLocaleString("zh-Hans-CN-u-nu-hanidec"));
-// 一二三,四五六,七八九,一二三,四五六,七八九
+// 一二三，四五六，七八九，一二三，四五六，七八九
```
The same as the above comment, you can try this case.

yin1999 commented on files/en-us/web/javascript/reference/global_objects/intl/datetimeformat/formattoparts/index.md:
````diff
@@ -206,7 +206,7 @@ commonly present these side-by-side:
 ```js
 const df = new Intl.DateTimeFormat("zh-u-ca-chinese", { year: "numeric" });
-df.format(Date.UTC(2012, 11, 17, 3, 0, 42)); // 2012壬辰年
+df.format(Date.UTC(2012, 11, 17, 3, 0, 42)); // 2012 壬辰年
````
The same as the above comment, you can try this.

yin1999 commented on files/en-us/web/javascript/reference/global_objects/intl/numberformat/index.md:
```diff
@@ -77,7 +77,7 @@ console.log(new Intl.NumberFormat("en-IN").format(number));
 // the nu extension key requests a numbering system, e.g. Chinese decimal
 console.log(new Intl.NumberFormat("zh-Hans-CN-u-nu-hanidec").format(number));
-// 一二三,四五六.七八九
+// 一二三，四五六。七八九
```
same

yin1999 commented on files/en-us/web/javascript/reference/global_objects/intl/numberformat/numberformat/index.md:
```diff
@@ -341,7 +341,7 @@ new Intl.NumberFormat("de", {
   notation: "compact",
 }).format(987654321);
-// 9.9亿
+// 9.9 亿
```
same

yin1999 commented on files/en-us/web/javascript/reference/global_objects/number/tolocalestring/index.md:
```diff
@@ -99,7 +99,7 @@ console.log(number.toLocaleString("en-IN"));
 // the nu extension key requests a numbering system, e.g. Chinese decimal
 console.log(number.toLocaleString("zh-Hans-CN-u-nu-hanidec"));
-// 一二三,四五六.七八九
+// 一二三，四五六。七八九
```
same

ihavecoke committed:
Standardize writing format for CJK languages

In accordance with the guidelines for Chinese copywriting, this commit adjusts the text formatting to adhere to the standard conventions for CJK (Chinese, Japanese, Korean) languages.
This update ensures consistency and improves the overall readability and clarity of the text.

ihavecoke commented:
Thank you @yin1999 @Josh-Cena , for taking the time to review my pull request. 
Your suggestions were valuable and made a lot of sense. 
I have now made the necessary adjustments to the code based on your recommendations. 
I kindly request you to review it once again. Thank you.

ihavecoke committed:
Merge branch 'main' into main

dipikabh commented:
cc @hochan222, @mfuji09 - Could you or someone from your teams please review the changes in this PR? Thank you!

hochan222 commented on files/jsondata/L10n-SVG.json:
```diff
@@ -103,7 +103,7 @@
     "fr": "N'importe quelle quantité d'éléments descriptifs, {{SVGElement(\"script\")}} et de deux à quatre éléments {{SVGElement(\"stop\")}}.",
     "ja": "任意の数の説明的要素、 {{SVGElement(\"script\")}} および 2 から 4 個の {{SVGElement(\"stop\")}} 要素。",
-    "ko": "임의의 개수의 설명 요소, {{SVGElement(\"script\")}}, 그리고 2개에서 4개의 {{SVGElement(\"stop\")}} 요소.",
+    "ko": "임의의 개수의 설명 요소, {{SVGElement(\"script\")}}, 그리고 2 개에서 4 개의 {{SVGElement(\"stop\")}} 요소.",
```
```suggestion
    "ko": "임의의 개수의 설명 요소, {{SVGElement(\"script\")}}, 그리고 2개에서 4개의 {{SVGElement(\"stop\")}} 요소.",
```

hochan222 commented on files/jsondata/L10n-SVG.json:
```diff
@@ -114,7 +114,7 @@
     "fr": "N'importe quelle quantité d'éléments d'animation ou d'éléments descriptifs ou d'éléments {{SVGElement(\"mask\")}}, {{SVGElement(\"clipPath\")}}, {{SVGElement(\"marker\")}}, {{SVGElement(\"script\")}}, et {{SVGElement(\"style\")}}, ainsi que zéro ou un élément {{SVGElement(\"meshgradient\")}}.",
     "ja": "任意の数のアニメーション要素、説明的要素、 {{SVGElement(\"mask\")}}, {{SVGElement(\"clipPath\")}}, {{SVGElement(\"marker\")}}, {{SVGElement(\"script\")}}, {{SVGElement(\"style\")}} の各要素と 0 個または 1 個の {{SVGElement(\"meshgradient\")}} 要素。",
-    "ko": "임의의 개수의 애니메이션 요소, 설명 요소, {{SVGElement(\"mask\")}}, {{SVGElement(\"clipPath\")}}, {{SVGElement(\"marker\")}}, {{SVGElement(\"script\")}}, 그리고 {{SVGElement(\"style\")}} 요소, 그리고 0개 또는 1개의 {{SVGElement(\"meshgradient\")}} 요소.",
+    "ko": "임의의 개수의 애니메이션 요소, 설명 요소, {{SVGElement(\"mask\")}}, {{SVGElement(\"clipPath\")}}, {{SVGElement(\"marker\")}}, {{SVGElement(\"script\")}}, 그리고 {{SVGElement(\"style\")}} 요소, 그리고 0 개 또는 1 개의 {{SVGElement(\"meshgradient\")}} 요소.",
```
```suggestion
    "ko": "임의의 개수의 애니메이션 요소, 설명 요소, {{SVGElement(\"mask\")}}, {{SVGElement(\"clipPath\")}}, {{SVGElement(\"marker\")}}, {{SVGElement(\"script\")}}, 그리고 {{SVGElement(\"style\")}} 요소, 그리고 0개 또는 1개의 {{SVGElement(\"meshgradient\")}} 요소.",
```

hochan222 commented on files/jsondata/L10n-CSS.json:
```diff
@@ -37,7 +37,7 @@
     "fr": "Valeur introuvable dans la base de données",
     "ja": "DB に値が見つかりません！",
-    "ko": "DB에 값이 없습니다!",
+    "ko": "DB 에 값이 없습니다!",
```
```suggestion
    "ko": "DB에 값이 없습니다!",
```

hochan222 reviewed (changes requested):
Hello. In Korean, the content before the change is common and there are no grammatical problems.

> 2 개에서 4 개의
> 그리고 0 개 또는 1 개의

When expressing quantity and order in the combination of "number + unit noun", spacing is the rule, but if the number comes(not word), pasting is also allowed. In Korea, spaces are usually not used when numbers appear. It is a common.

- (bad) `한개`
- (good) `한 개`
- (good, more common) `1개`
- (good) `1 개`

> "DB 에 값이 없습니다!"

There should be no spaces.

- (bad) `DB 에`
- (good) `DB에`

mfuji09 commented on files/en-us/web/css/writing-mode/index.md:
```diff
@@ -91,35 +91,35 @@ The HTML is a {{HTMLElement("table")}} with each writing mode in a row with a co
     <td>Example text</td>
     <td>מלל ארוך לדוגמא</td>
-    <td>1994年に至っては</td>
+    <td>1994 年に至っては</td>
```
These examples are for mixed writing-mode. In Japanese, half-width spaces isn't used between Japanese characters and Latin letters in vertical / mixed-writing mode. So these examples intentionally omit spaces around numbers.

```suggestion
    <td>1994年に至っては</td>
```

mfuji09 commented on files/en-us/web/css/writing-mode/index.md:
```diff
@@ -91,35 +91,35 @@ The HTML is a {{HTMLElement("table")}} with each writing mode in a row with a co
     <td>Example text</td>
     <td>מלל ארוך לדוגמא</td>
-    <td>1994年に至っては</td>
+    <td>1994 年に至っては</td>
```
```suggestion
    <td>1994年に至っては</td>
```

mfuji09 commented on files/en-us/web/css/writing-mode/index.md:
```diff
@@ -91,35 +91,35 @@ The HTML is a {{HTMLElement("table")}} with each writing mode in a row with a co
     <td>Example text</td>
     <td>מלל ארוך לדוגמא</td>
-    <td>1994年に至っては</td>
+    <td>1994 年に至っては</td>
```
```suggestion
    <td>1994年に至っては</td>
```

mfuji09 commented on files/en-us/web/css/writing-mode/index.md:
```diff
@@ -91,35 +91,35 @@ The HTML is a {{HTMLElement("table")}} with each writing mode in a row with a co
     <td>Example text</td>
     <td>מלל ארוך לדוגמא</td>
-    <td>1994年に至っては</td>
+    <td>1994 年に至っては</td>
```
```suggestion
    <td>1994年に至っては</td>
```

mfuji09 commented on files/en-us/web/css/writing-mode/index.md:
```diff
@@ -91,35 +91,35 @@ The HTML is a {{HTMLElement("table")}} with each writing mode in a row with a co
     <td>Example text</td>
     <td>מלל ארוך לדוגמא</td>
-    <td>1994年に至っては</td>
+    <td>1994 年に至っては</td>
```
```suggestion
    <td>1994年に至っては</td>
```

mfuji09 commented on files/jsondata/L10n-Common.json:
```diff
@@ -190,7 +190,7 @@
     "es": "Nuestros voluntarios aún no han traducido este artículo al Español. ¡Únete a nosotros y ayuda a traducirlo!",
     "fr": "Nos volontaires n'ont pas encore traduit cet article en français. Rejoignez-nous et aidez-nous à le faire!",
-    "ja": "まだボランティアによって日本語に翻訳されていない記事です。 MDN に参加して、翻訳してみませんか？",
+    "ja": "まだボランティアによって日本語に翻訳されていない記事です。MDN に参加して、翻訳してみませんか？",
```
```suggestion
    "ja": "まだボランティアによって日本語に翻訳されていない記事です。 MDN に参加して、翻訳してみませんか？",
```

mfuji09 commented on files/en-us/web/css/text-decoration-skip-ink/index.md:
```diff
@@ -56,9 +56,9 @@ text-decoration-skip-ink: unset;
 <p>You should go on a quest for a cup of coffee.</p>
 <p class="no-skip-ink">Or maybe you'd prefer some tea?</p>
-<p>この文は、 text-decoration-skip-ink: auto の使用例を示しています。</p>
+<p>この文は、text-decoration-skip-ink: auto の使用例を示しています。</p>
```
```suggestion
<p>この文は、 text-decoration-skip-ink: auto の使用例を示しています。</p>
```

mfuji09 commented on files/en-us/web/css/text-decoration-skip-ink/index.md:
```diff
@@ -56,9 +56,9 @@ text-decoration-skip-ink: unset;
+<p>この文は、text-decoration-skip-ink: auto の使用例を示しています。</p>
 <p class="skip-ink-all">
-  この文は、 text-decoration-skip-ink: all の使用例を示しています。
+  この文は、text-decoration-skip-ink: all の使用例を示しています。
```
```suggestion
  この文は、 text-decoration-skip-ink: all の使用例を示しています。
```

mfuji09 reviewed (changes requested):
This copywriting guidelines isn't for Japanese, and it isn't accepted from Japanese writers.
There are a lot of differences in writing behavior between Chinese and Japanese (or, there is few common things.)
We (Japanese community) already have an own writing guideline, so please let us (native Japanese writers) to decide how to write.

Josh-Cena reviewed (commented):
After applying the reviews for both JA and KO, there will be no changes left. Therefore I'll just go ahead and close it. Thanks for the reviews!

dipikabh commented:
Thank you, @hochan222 and @mfuji09, for your prompt reviews and feedback!
